spec/lib/soapy_cake/client_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+RSpec.describe SoapyCake::Client do
+  subject(:client) { described_class.new }
+
+  let(:http) do
+    instance_double(Net::HTTP)
+  end
+
+  let(:request) do
+    SoapyCake::Request.new(:admin, :addedit, :advertiser, {})
+  end
+
+  describe '#run' do
+    before do
+      allow(Net::HTTP)
+        .to receive(:start)
+        .and_yield(http)
+      allow(http)
+        .to receive(:request)
+        .and_return(response)
+    end
+
+    context 'receives non-200 HTTP status code' do
+      let(:response) do
+        instance_double(Net::HTTPInternalServerError, code: 500, body: '🔥')
```

Comment:
How about [webmock](response)?

Reply:
I'm just testing the logic between the two places a `RequestFailed` exception is raised and the information is attached. Integration tests already exist.

Reply:
But webmock does just the same what you do here, and wraps it in a better looking DSL.

Reply:
I'm not sure what you mean. Your link is broken.

Reply:
Sorry, too much copy-pasting. This link https://rubygems.org/gems/webmock

Reply:
Ah! Good idea! Updated!